math/src/polynomial/multilinear_term.rs

```diff
@@ -0,0 +1,183 @@
+use crate::field::element::FieldElement;
+use crate::field::traits::IsField;
+use crate::polynomial::term::Term;
+
+/// Struct for (coeff: FieldElement<F>, terms: Vec<usize>) representing a multilinear
+/// monomial in a sparse format.
+// TODO: add check that var labels are 0 indexed
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct MultiLinearMonomial<F: IsField>
+where
+    <F as IsField>::BaseType: Send + Sync,
+{
+    pub coeff: FieldElement<F>,
+    pub vars: Vec<usize>,
+}
+
+impl<F: IsField> MultiLinearMonomial<F>
+where
+    <F as IsField>::BaseType: Send + Sync,
+{
+    /// Create a new `Term` from a tuple of the form `(coeff, (variables))`
+    pub fn new(term: (FieldElement<F>, Vec<usize>)) -> Self {
+        // sort variables in increasing order
+        let mut vars = term.1;
+        vars.sort();
+
+        Self {
+            coeff: term.0,
+            vars,
+        }
+    }
+}
+
+impl<F: IsField> Term<F> for MultiLinearMonomial<F>
+where
+    <F as IsField>::BaseType: Send + Sync,
+{
+    /// Returns the total degree of `self`. This is the count of all variables
+    fn degree(&self) -> usize {
+        self.vars.len()
+    }
+
+    /// Returns a list of the powers of each variable in `self` i.e. numbers representing the id of
+    /// the specific variable
+    fn vars(&self) -> Vec<usize> {
+        self.vars.clone()
+    }
+
+    fn powers(&self) -> Vec<usize> {
+        vec![1; self.vars.len()]
+    }
+
+    /// Fetches the max variable by id from the sparse list of id's this is used to ensure the upon
+    /// evaluation the correct number of points are supplied
+    fn max_var(&self) -> usize {
+        // Sparse variables are stored in increasing var_id therefore we grab the last one
+        match self.vars.last() {
+            Some(&max_var) => max_var,
+            None => 0,
+        }
+    }
+
+    // TODO: test this
+    /// Evaluates `self` at the point `p`.
+    fn evaluate(&self, p: &[FieldElement<F>]) -> FieldElement<F> {
+        // Check that p contains the proper amount of elements in dense form.
+        //assert!(self.max_var() < p.len());
+
+        if self.vars.is_empty() || p.is_empty() {
+            return self.coeff.clone();
+        }
+
+        // var_id is index of p
+        let eval = self
+            .vars
+            .iter()
+            .fold(FieldElement::<F>::one(), |acc, x| acc * p[*x].clone());
```

Comment:
You should be able to remove this clone. The multiplication should take a reference.